### PR TITLE
Fix Textarea when providing empty line, and prevent broken settings page

### PR DESCRIPTION
**Pull request by alcalyn**
### Description:

When addind an empty line in a TextArea in "General settings" page, i.e in ips whitelist:

```
1.1.1.1

2.2.2.2
```

The array value is stored as `{"0": "1.1.1.1", "2": "2.2.2.2"}` because of the array_filter.

Also the `validate` is called before the `transform`, which seems to be expected, but makes the error uncached.

The impact is the "General settings" page throw an error and stop rendering, so all plugin settings after the error are not rendered, then the page does not display all plugins settings. This PR also catch this js error in case other plugins have the same error, so that all plugins settings are displayed.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CorePluginsAdmin/vue/src/FormField/FieldTextareaArray.vue [hunk 1]
```diff
@@ -46,7 +46,18 @@ export default defineComponent({
...
+      if (typeof this.modelValue === 'object') {
+        return Object.values(this.modelValue).join(SEPARATOR);
+      }
+
+      try {
+        return (this.modelValue || []).join(SEPARATOR);
+      } catch (e) {
+        // Prevent page breaking on unexpected modelValue type
+        console.error(e);
+        return [];
```
Comment (sgiehl) [inline plugins/CorePluginsAdmin/vue/src/FormField/FieldTextareaArray.vue]: The method is supposed to return a string.

```suggestion
        return '';
```

**Review by sgiehl — CHANGES_REQUESTED** [review]
Thanks @alcalyn for contributing this PR. That's very appreciated.
I was able to reproduce the problem you described. Your fixes should indeed solved that 💯 
I left one suggestion, that needs to be applied. In addition we have commited the built vue files to git. So to finish the PR it would be needed to built the files.
For that you may need to run an `npm install` to fetch the related dependencies. Afterwards it should be possible to run `./console vue:build` to build the vue files.
If you aren't able to build the files, let me know. I can also do that once the PR was merged.

File: plugins/CorePluginsAdmin/vue/src/FormField/FieldTextareaArray.vue (hunk 1 above)
Comment (alcalyn) [reply plugins/CorePluginsAdmin/vue/src/FormField/FieldTextareaArray.vue]: Fixed

**Comment by alcalyn** [comment]
Thanks for the help, fixed it and vue files rebuilt.

**Review by sgiehl — APPROVED** [review]
Thanks for updating the PR. Works as expected now.

**Pull request merged**